r/clean-enhanced-trace-with-r.qmd

```diff
@@ -43,99 +43,75 @@ clean_enhanced_trace <- function(cusips,
   if (!dbIsValid(connection)) stop("Connection issue.")
 
   # Enhanced Trace ----------------------------------------------------------
+  trace_enhanced_db <- tbl(connection, I("trace.trace_enhanced"))
+  
   # Main file
-  trace_all <- tbl(
-    connection,
-    I("trace.trace_enhanced")
-  ) |>
-    filter(cusip_id %in% cusips) |>
-    filter(trd_exctn_dt >= start_date & trd_exctn_dt <= end_date) |>
-    select(
-      cusip_id, msg_seq_nb, orig_msg_seq_nb,
-      entrd_vol_qt, rptd_pr, yld_pt, rpt_side_cd, cntra_mp_id,
-      trd_exctn_dt, trd_exctn_tm, trd_rpt_dt, trd_rpt_tm,
-      pr_trd_dt, trc_st, asof_cd, wis_fl,
-      days_to_sttl_ct, stlmnt_dt, spcl_trd_fl
+   trace_all <- trace_enhanced_db |>
+    filter(
+      cusip_id %in% cusips,
+      between(trd_exctn_dt, start_date, end_date)
     ) |>
+    select(cusip_id, msg_seq_nb, orig_msg_seq_nb,
+           entrd_vol_qt, rptd_pr, yld_pt, rpt_side_cd, cntra_mp_id,
+           trd_exctn_dt, trd_exctn_tm, trd_rpt_dt, trd_rpt_tm,
+           pr_trd_dt, trc_st, asof_cd, wis_fl,
+           days_to_sttl_ct, stlmnt_dt, spcl_trd_fl) |>
     collect()
 
   # Enhanced Trace: Post 06-02-2012 -----------------------------------------
   # Trades (trc_st = T) and correction (trc_st = R)
   trace_post_TR <- trace_all |>
-    filter(
-      (trc_st == "T" | trc_st == "R"),
-      trd_rpt_dt >= as.Date("2012-02-06")
-    )
+    filter((trc_st == "T" | trc_st == "R"),
+           trd_rpt_dt >= as.Date("2012-02-06"))
 
-  # Cancellations (trc_st = X) and correction cancellations (trc_st = C)
+  # Cancelations (trc_st = X) and correction cancelations (trc_st = C)
   trace_post_XC <- trace_all |>
-    filter(
-      (trc_st == "X" | trc_st == "C"),
-      trd_rpt_dt >= as.Date("2012-02-06")
-    )
+    filter((trc_st == "X" | trc_st == "C"),
+           trd_rpt_dt >= as.Date("2012-02-06"))
 
   # Cleaning corrected and cancelled trades
   trace_post_TR <- trace_post_TR |>
     anti_join(trace_post_XC,
-      join_by(
-        cusip_id, msg_seq_nb, entrd_vol_qt,
-        rptd_pr, rpt_side_cd, cntra_mp_id,
-        trd_exctn_dt, trd_exctn_tm
-      )
-    )
+              by = c("cusip_id", "msg_seq_nb", "entrd_vol_qt",
```

Comment:
maybe I missed something but we agreed to consistently use join_by for joins?

Reply:
Yes, seems like we missed it in the package. I copied the code from there.